src/FrameDecorator/AbstractFrameDecorator.php

```diff
@@ -694,29 +707,33 @@ public function split(Frame $child = null, bool $force_pagebreak = false)
             $node->removeAttribute("id");
         }
 
+        $this->revert_counter_increment();
         $split = $this->copy($node->cloneNode());
-        $split->reset();
-        $split->get_original_style()->text_indent = 0;
-        $split->_splitted = true;
-        $split->_already_pushed = true;
 
-        // The body's properties must be kept
+        $style = $this->_frame->get_style();
+        $split_style = $split->get_original_style();
+
+        // Truncate the box decoration at the split, except for the body
         if ($node->nodeName !== "body") {
             // Style reset on the first and second parts
-            $style = $this->_frame->get_style();
             $style->margin_bottom = 0;
             $style->padding_bottom = 0;
             $style->border_bottom = 0;
 
             // second
-            $orig_style = $split->get_original_style();
-            $orig_style->text_indent = 0;
-            $orig_style->margin_top = 0;
-            $orig_style->padding_top = 0;
-            $orig_style->border_top = 0;
-            $orig_style->page_break_before = "auto";
+            $split_style->margin_top = 0;
+            $split_style->padding_top = 0;
+            $split_style->border_top = 0;
+            $split_style->page_break_before = "auto";
         }
 
+        $split_style->text_indent = 0;
+        $split_style->counter_reset = "none";
+
+        $split->set_style(clone $split_style);
```

Comment:
This is what `$split->reset()` effectively did. All the other resets are unnecessary because a newly decorated frame is created for the split node.